### DISCUSSION TITLE
PY-UVM Framework for RISC-V Single Cycle Core

### DISCUSSION THREAD

**Discussion by merldsu** [raysalemi/Python4RTLVerification#8] [Show and tell]
Hello,

We are from Microelectronics Research Lab (MERL-DSU). MERL is an non-profitable organization with an ambitious plan to lead Microelectronics Research & Development in Pakistan. MERL is working with a Vision to enable Pakistan to become a recognized global player in the microelectronics industry. MERL's Mission is to train Undergraduate Students of Pakistan in the field of ASIC designing. 

As we all know, one of the most important steps in ASIC design is verification. Verification may be done in a variety of ways utilising commercial tools. As the opensource community is growing, opensource tools are becoming increasingly popular.

So to contribute in an opensource community our verification team have designed an Initial PY-UVM Framework for RISC-V Single Cycle Core. Our framework is able to verify the RV32IM supported designs based on RISC-V ISA. 

More details of our framework can be found on the following github

[https://github.com/merldsu/Py_UVM_Framework.git](https://github.com/merldsu/Py_UVM_Framework.git)

Thank you. We are looking forward to hearing from you and would like to collaborate to enhance the design framework.
 
Regards:
MERL-DSU

Contact info:
Email: merldsu2@gmail.com
Facebook: [https://www.facebook.com/merldsu](https://www.facebook.com/merldsu)
Youtube: [https://www.youtube.com/@merldsu](https://www.youtube.com/@merldsu)

**Comment by raysalemi** [comment]
Fascinating! I am glad to see that **pyuvm** was helpful to this effort.

**Reply by merldsu** [reply]
Thank you for your quick response. We would like you to review our framework and share your valuable feedback.

**Reply by raysalemi** [reply]
Hi. I am afraid that my time is completely taken by teaching high school. I retired from Siemens and became a teacher.  

If you tag me on a post on LinkedIn,  would be happy to reshare it. Perhaps others could review the framework.

**Reply by merldsu** [reply]
Sure will do that. Thanks
